Overhaul for modern python+mopidy

djmattyg007 opened the pull request:
- Support python3
- Use standard mopidy code quality tools
- Replace travis with github actions
- Rewrite tests to use pytest
- Add initial type checking
- Modify configuration handling slightly
- Other minor tweaks
- Update readme
  - Reformat to make it easier to read outside of github
  - Update links
  - Remove changelog from readme. It's of little value in its current
    form, and should be continued as a standalone file if still deemed
    necessary in future.

djmattyg007 committed:
Overhaul for modern python+mopidy

- Support python3
- Use standard mopidy code quality tools
- Replace travis with github actions
- Rewrite tests to use pytest
- Add initial type checking
- Mopidy configuration handling slightly
- Other minor tweaks
- Update readme
  - Reformat to make it easier to read outside of github
  - Update links
  - Remove changelog from readme. It's of little value in its current
    form, and should be continued as a standalone file if still deemed
    necessary in future.

djmattyg007 commented:
A couple of notes:

- Releases to PyPI should happen automatically when a Github Release is created based on a tag. So once this is merged, you'll need to create a tag for `1.0.0`, push it to Github, then create a Github release.
- Before you do that, you'll need to add a secret to this repo with the label `PYPI_TOKEN`. You'll need to generate this token on PyPI, and ideally it should be scoped to just `Mopidy-Webhooks`. This is compulsory for performing releases.
- I've set up the repo with a tool named `invoke`, which is a python task runner. In case you haven't come across it before, you can find some docs here: http://docs.pyinvoke.org/ I've found it to be a great way of scripting common tasks in python repos.

djmattyg007 commented:
Also it looks like you'll need to enable Github Actions on this repo so that the workflows can run at all.

djmattyg007 commented on README.rst:
```diff
@@ -1,27 +1,29 @@
+    :target: https://github.com/paddycarey/mopidy-webhooks/actions?query=branch%3Amaster+workflow%3ACI
+    :alt: Github Actions CI build status
 
 .. image:: https://img.shields.io/coveralls/paddycarey/mopidy-webhooks/master.svg?style=flat
    :target: https://coveralls.io/r/paddycarey/mopidy-webhooks?branch=master
    :alt: Test coverage
```
This probably needs updating to point at codecov, but you'd need to sign up there to get the coverage data flowing.

paddycarey replied:
Let's remove codecov for now, and once the action is running post-merge we can add it back (just trying to minimise the number of new things being merge at once).

djmattyg007 commented on mopidy_webhooks/reporters/status.py:
```diff
@@ -22,39 +17,46 @@ class StatusReporter(pykka.ThreadingActor):
+        playback_state_future = self.core.playback.get_state()
+        current_track_future = self.core.playback.get_current_track()
+        time_position = self.core.playback.get_time_position()
+
+        playback_state = playback_state_future.get()
+
         current_status = {
-            'current_track': self.core.playback.current_track.get(),
-            'state': self.core.playback.state.get(),
-            'time_position': self.core.playback.time_position.get(),
+            "state": playback_state,
+            "current_track": current_track_future.get(),
+            "time_position": time_position.get(),
         }
-        send_webhook(self.config, {'status_report': current_status})
-        self.report_again(current_status)
+        payload = {
+            "type": "status",
+            "data": current_status,
+        }
+
+        send_webhook(self.config, payload)
+        self.report_again(playback_state)
```
Hopefully when this PR is merged, this code can either be replaced with a single call to `core.get_status()`, or a second function named something like "full status" could be added to do that instead.

https://github.com/mopidy/mopidy/pull/2007

paddycarey commented:
> Also it looks like you'll need to enable Github Actions on this repo so that the workflows can run at all.

I checked and they're already enabled, do they need enabled in your forked repo perhaps?

paddycarey commented on setup.cfg:
```diff
@@ -1,2 +1,92 @@
+[metadata]
+name = Mopidy-Webhooks
+version = 1.0.0
+url = https://github.com/djmattyg007/mopidy-webhooks
```
should this still point at the source repo?

djmattyg007 replied:
It should 😅

paddycarey reviewed (commented):
Sorry for the delay, I've been on vacation. Just one small comment, but otherwise looks great, thanks!

Let me figure out how to make the tests run and then we can get it merged and released.

djmattyg007 commented:
I've updated the PR. Hopefully it's good to go now.

I suspect the reason the CI workflow isn't running yet is because of some changes Github made recently to prevent abuse of Github Actions. As I haven't contributed to this repo before, and the workflow didn't exist before this PR, I doubt it will run.

I think it will be easier to merge this PR first, then worry if the workflow doesn't start running after that. If it doesn't work, I'll be more than happy to submit another PR to make it work :)

paddycarey reviewed (approved):
LGTM (once the codecov comment above is addressed, will merge once that's taken care of)

djmattyg007 commented:
I've updated the PR with the following changes:

- Removed the codecov step from the CI workflow
- Removed the codecov badge from the readme
- Removed the old URL target for the CI badge in the readme

paddycarey commented:
@djmattyg007 merged, thanks! sorry for the delay in responses here, been a busy few weeks 😅

paddycarey commented:
looks like there were a few warnings in the actions run, but it completed successfully https://github.com/paddycarey/mopidy-webhooks/actions/runs/1214740261

djmattyg007 commented:
I've been seeing similar warnings with Python 3.10 in other repos. Not sure what's going on :shrug: But 3.10-rc2 came out recently, so I'll submit another PR to update the workflow and I guess we'll see if that fixes it.
